src/main/java/seedu/address/model/guest/GuestTagsContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.guest;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Guest}'s {@code tags} exactly matches a single {@code tag} keyword argument.
+ */
+public class GuestTagsContainsKeywordsPredicate implements Predicate<Guest> {
+    private final List<String> keywords;
+
+    public GuestTagsContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Guest guest) {
+        List<Tag> guestTags = new LinkedList<>();
+        guestTags.addAll(guest.getTags());
+
+        for (String keywordTag : keywords) {
+            for (Tag guestTag : guestTags) {
+                if (guestTag.tagName.equals(keywordTag)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof GuestTagsContainsKeywordsPredicate // instanceof handles nulls
+                && keywords.equals(((GuestTagsContainsKeywordsPredicate) other).keywords)); // state check
+    }
```

Comment:
I think it's standard practice to override hashCode whenever you override equals.